### PR TITLE
iptables: Reorder comment postition

### LINKED ISSUE #71444: iptables comment parameter incorrectly positioned in iptables
##### SUMMARY
When using **comment** parameter with iptables module, it is not positioned as the right-most object in the actual iptables table.
The outcome looks very bad at the moment.

##### ISSUE TYPE
- Cosmetic Bug Report

##### COMPONENT NAME

**iptables** module, **comment** parameter

##### ANSIBLE VERSION

```paste below
ansible 2.9.9
  config file = /home/me/project/ansible.cfg
  configured module search path = ['/home/me/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/me/vConstruction/lib/python3.6/site-packages/ansible
  executable location = /home/me/vConstruction/bin/ansible
  python version = 3.6.9 (default, Jul 17 2020, 12:50:27) [GCC 8.4.0]
```

##### CONFIGURATION

```paste below
DEFAULT_CALLBACK_WHITELIST(/home/me/project/ansible.cfg) = ['profile_tasks']
DEFAULT_HOST_LIST(/home/me/project/ansible.cfg) = ['/home/me/project/inventory']
DEPRECATION_WARNINGS(/home/me/project/ansible.cfg) = False
HOST_KEY_CHECKING(/home/me/project/ansible.cfg) = False
```

##### OS / ENVIRONMENT

Host where I use Ansible: Ubuntu 18.04 via WSL 2.
Target VM where the iptables rules are being created: Ubuntu 18.04.

##### STEPS TO REPRODUCE

It can be reproduced by using the following task and running `iptables -L` on the target machine.
I added the DNS task as well, because this one looks correct, but I'm sure that's just because I don't have **ctstate** specified.

```yaml
- name: Create HTTPS rule
  iptables:
    chain: INPUT
    in_interface: eth0
    protocol: tcp
    destination_port: https
    ctstate: NEW
    jump: ACCEPT
    comment: Accept new inbound HTTPS connections to eth0 interface

- name: Create DNS rule
  iptables:
    chain: INPUT
    protocol: udp
    destination_port: domain
    jump: ACCEPT
    comment: Accept inbound DNS connections
```

##### EXPECTED RESULTS

```bash
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:ssh  ctstate NEW /* Accept new inbound HTTPS connections to eth0 interface */
```
##### ACTUAL RESULTS

```bash
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:https /* Accept new inbound HTTPS connections to eth0 interface */ ctstate NEW
ACCEPT     udp  --  anywhere             anywhere             udp dpt:domain /* Accept inbound DNS connections */
```

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes: #71444.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
iptables

### CODE REVIEW

**Comment by aminvakil** [comment]
ready_for_review

**Comment by bcoca** [comment]
needs a changelog

**Comment by aminvakil** [comment]
ready_for_review

File: test/units/modules/test_iptables.py [hunk 1]
```diff
@@ -876,3 +876,39 @@ def test_insert_rule_with_wait(self):
...
             'ACCEPT'
         ])
+
+    def test_comment_position_at_end(self):
+        """Test flush without parameters"""
+        set_module_args({
+            'chain': 'INPUT',
+            'jump': 'ACCEPT',
+            'action': 'insert',
+            'comment': 'this is a comment',
```
Comment (samdoran) [inline test/units/modules/test_iptables.py:887]: I think this should include some parameters that are problematic when they appear before `comment`.
```suggestion
            'cstate': ['NEW'],
            'comment': 'this is a comment',
```

File: test/units/modules/test_iptables.py [hunk 2]
```diff
@@ -876,3 +876,39 @@ def test_insert_rule_with_wait(self):
...
+
+        self.assertEqual(run_command.call_count, 1)
+        self.assertEqual(run_command.call_args_list[0][0][0], [
+            '/sbin/iptables',
+            '-t',
+            'filter',
+            '-C',
+            'INPUT',
+            '-j',
+            'ACCEPT',
```
Comment (samdoran) [inline test/units/modules/test_iptables.py:909]:
```suggestion
            'ACCEPT',
            '--cstate',
            'NEW',
```

**Review by samdoran — CHANGES_REQUESTED** [review]
I tested this fix and it is correct. Just some minor changes to the tests.

File: test/units/modules/test_iptables.py (hunk 1 above)
Comment (aminvakil) [reply test/units/modules/test_iptables.py:887]: Fixed in https://github.com/ansible/ansible/pull/71496/files#diff-a184f210bf607737e23e4930d925ba5fR886.

File: test/units/modules/test_iptables.py (hunk 2 above)
Comment (aminvakil) [reply test/units/modules/test_iptables.py:909]: Fixed in https://github.com/ansible/ansible/pull/71496/files#diff-a184f210bf607737e23e4930d925ba5fR910-R913.

**Comment by aminvakil** [comment]
ready_for_review

**Review by samdoran — APPROVED** [review]

**Comment by samdoran** [comment]
rebuild_merge

**Pull request merged**